### PR TITLE
mgr/dashboard: Refresh button on the iscsi targets page

### PR DESCRIPTION
**Pull request by nizamial09**
Added a refresh button on the iscsi targets page. I've noticed that the
auto reload causes some load on the backend. So I disabled the auto
reload and went for the same approach as we have on rgw. A yellow
warning color on refresh btn to let the user know to manually refresh
that whenever needed.

Screenshot from 2021-08-09 13-00-33

Fixes: https://tracker.ceph.com/issues/52102
Signed-off-by: Nizamudeen A <nia@redhat.com>

## Checklist
- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.ts [hunk 1]
```diff
@@ -124,29 +128,49 @@ export class IscsiTargetListComponent extends ListWithDetails implements OnInit,
...
-          this.settings = settings;
-        });
-      } else {
+      if (!result.available) {
         this.status = result.message;
       }
     });
   }
 
+  timeConditionReached() {
```
Comment (alfonsomthd) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.ts:137]: Following on our clean code DRY agreement and taking int oaccount tahta we have already have 2 identical `timeConditionReached` declarations, please extract this method for reusability (i.e. declaring it in: src/pybind/mgr/dashboard/frontend/src/app/shared/classes/list-with-details.class.ts)

**Review by alfonsomthd — CHANGES_REQUESTED** [review]

**Review by epuertat — COMMENTED** [review]
@nizamial09 you mention that auto-reloading would cause "some load". Can you please be more specific? Have you done any estimation about that? Have you noticed any scalability issue with a large number of gateways/targets/etc.?

In general, the manual refresh approach is not desirable and we should keep it to the minimum, only when there are justified reasons for that. An alternative approach, if needed, might be to increase the polling period (from 5 to 10, 20 secs). That's always better than `inf` seconds.

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.ts (hunk 1 above)
Comment (nizamial09) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-list/iscsi-target-list.component.ts]: Done.

**Comment by nizamial09** [comment]
jenkins test dashboard cephadm

**Comment by nizamial09** [comment]
jenkins test dashboard

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard

**Comment by alfonsomthd** [comment]
jenkins test dashboard

**Review by alfonsomthd — APPROVED** [review]

**Comment by nizamial09** [comment]
jenkins test dashboard

**Review by avanthakkar — APPROVED** [review]
LGTM!

**Comment by alfonsomthd** [comment]
jenkins test dashboard cephadm

**Comment by alfonsomthd** [comment]
jenkins test dashboard

**Pull request merged**